src/SFA.DAS.Tools.Servicebus.Support.Infrastructure/Services/CosmosDbContext.cs

```diff
@@ -78,7 +80,7 @@ public async Task DeleteQueueMessagesAsync(IEnumerable<string> ids)
 
         public async Task<IEnumerable<QueueMessage>> GetQueueMessagesAsync(string userId, SearchProperties searchProperties)
         {
-            var sqlQuery = $"SELECT * FROM c WHERE c.userId ='{userId}'";
+            var sqlQuery = $"SELECT * FROM c WHERE c.userId ='{userId}' and c.type='message'";
```

Comment:
How does this affect performance?

Reply:
From my understanding there are no performance implications. It is one of the recommended ways to create relationship